src/sentry/incidents/grouptype.py

```diff
@@ -45,6 +52,79 @@ class MetricIssueEvidenceData(EvidenceData[float]):
 MetricResult = int | dict
 
 
+class SessionsAggregate(StrEnum):
+    CRASH_FREE_SESSIONS = "percentage(sessions_crashed, sessions) AS _crash_rate_alert_aggregate"
+    CRASH_FREE_USERS = "percentage(users_crashed, users) AS _crash_rate_alert_aggregate"
+
+
+ALERT_TYPE_IDENTIFIERS: dict[Dataset, dict[str, str]] = {
+    Dataset.Events: {"num_errors": "count()", "users_experiencing_errors": "count_unique(user)"},
+    Dataset.Transactions: {
+        "throughput": "count()",
+        "trans_duration": "transaction.duration",
+        "apdex": "apdex",
+        "failure_rate": "failure_rate()",
+        "lcp": "measurements.lcp",
+        "fid": "measurements.fid",
+        "cls": "measurements.cls",
+    },
+    Dataset.PerformanceMetrics: {
+        "throughput": "count()",
+        "trans_duration": "transaction.duration",
+        "apdex": "apdex",
+        "failure_rate": "failure_rate()",
+        "lcp": "measurements.lcp",
+        "fid": "measurements.fid",
+        "cls": "measurements.cls",
+    },
+    Dataset.Sessions: {
+        "crash_free_sessions": SessionsAggregate.CRASH_FREE_SESSIONS.value,
+        "crash_free_users": SessionsAggregate.CRASH_FREE_USERS.value,
+    },
+    Dataset.Metrics: {
+        "crash_free_sessions": SessionsAggregate.CRASH_FREE_SESSIONS.value,
+        "crash_free_users": SessionsAggregate.CRASH_FREE_USERS.value,
+    },
+    Dataset.EventsAnalyticsPlatform: {
+        "trace_item_throughput": "count(span.duration)",
+        "trace_item_duration": "span.duration",
+        "trace_item_apdex": "apdex",
+        "trace_item_failure_rate": "failure_rate()",
+        "trace_item_lcp": "measurements.lcp",
+    },
+}
+
+
+def get_alert_type_from_aggregate_dataset(
+    aggregate: str, dataset: Dataset, organization: Organization | None = None
+) -> str:
+    """
+    Given an aggregate and dataset object, will return the corresponding wizard alert type
+    e.g. {'aggregate': 'count()', 'dataset': Dataset.ERRORS} will yield 'num_errors'
+
+    This function is used to format the aggregate value for anomaly detection issues.
+    """
+    identifier_for_dataset = ALERT_TYPE_IDENTIFIERS.get(dataset, {})
+
+    # Find matching alert type entry
+    matching_alert_type: None | str = None
```

Comment:
is there a better type we could use than `str`? could we make/use an enum or a set of literals instead? that way editors can do fun stuff like auto complete and we could catch any typo issues.